File: Sources/AgentStudio/Features/Bridge/Models/Telemetry/BridgeOperationCorrelation.swift
import AgentStudioInfrastructure
import CryptoKit
import Foundation

enum BridgeOperationCorrelation {
    /// Mints an owner-local UUIDv7 and returns only its scrubbed wire value.
    static func mintScrubbedID() -> String {
        let ownerLocalIdentity = UUIDv7.generate()
        return SHA256.hash(data: Data(ownerLocalIdentity.uuidString.utf8))
            .map { String(format: "%02x", $0) }
            .joined()
    }
}

File: BridgeWeb/src/core/comm-worker/bridge-operation-correlation.ts
import { uuidv7 } from 'uuidv7';

import { BridgeIncrementalSha256 } from './bridge-incremental-sha256.js';

const operationIdentityEncoder = new TextEncoder();

/**
 * Mints one owner-local UUIDv7 and returns only its scrubbed wire projection.
 * The raw identity never leaves this synchronous admission boundary.
 */
export function mintBridgeOperationCorrelationId(): string {
	const operationIdentity = uuidv7();
	const digest = new BridgeIncrementalSha256();
	digest.update(operationIdentityEncoder.encode(operationIdentity));
	return digest.digestHex();
}
